merlin/models/tf/core/combinators.py

```diff
@@ -615,7 +615,11 @@ def build(self, input_shape):
     def _maybe_filter_layer_inputs_using_schema(self, name, layer, inputs):
         maybe_schema = getattr(layer, "_schema", None)
         if maybe_schema and isinstance(inputs, dict):
-            layer_inputs = {k: v for k, v in inputs.items() if k in maybe_schema.column_names}
+            layer_inputs = {
+                k: v
+                for k, v in inputs.items()
+                if k.replace("__values", "").replace("__offsets", "") in maybe_schema.column_names
```

Comment:
what situation is this change required for. If we're handling the ragged inputs in the first layer how are the values/offsets making their way through to a ParallelBlock?

Reply:
You are right. In general we use `PrepareFeatures` in the `Model` right at the beginning to ensure that lists are converted from our ragged tensor convension (__values and __offsets) to `tf.RaggedTensor`.
But for `RetrievalModelV2` we cannot `PrepareFeatures` in `Model`, because in that case the `Encoders` (which are `tf.keras.Model`) would receive `tf.RaggedTensor` as inputs and that would cause problems when the query encoder is saved to be served on Triton.
That is why on `RetrievalModelV2` constructor we set `prep_features=False` and let the `Encoder` class to `prep_features=True`. As the `RetrievalModelV2` is a `ParallelBlock` of `Encoder` that is when it would receive the ragged convension (`__values` and `__offsets`) and this change is to ensure that the ragged representation is cascaded to the `Encoder` block